### ISSUE TITLE
[BUG]  check-unused-code doesn't detect stateful widgets

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#682]
* **Dart code metrics version:** 4.10.1
* **Dart sdk version:**  2.16.0 (stable)

**What did you do? Please include the source code example causing the issue.**
`flutter pub run dart_code_metrics:metrics check-unused-code lib --monorepo`

<details>
<summary>Code</summary>

```dart
import 'package:flutter/material.dart';

// LINT
class TestingSimpleClass {}

// LINT
class TestingStateless extends StatelessWidget { 
  const TestingStateless({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

 // DOES NOT LINT
class TestingStateFull extends StatefulWidget {
  const TestingStateFull({Key? key}) : super(key: key);

  @override
  _TestingStateFullState createState() => _TestingStateFullState();
}

class _TestingStateFullState extends State<TestingStateFull> {
  @override
  Widget build(BuildContext context) {
    return Container();
  }
}

```

</details>

**What did you expect to happen?**
TestingStateFull is reported as unused

**What actually happened?**
TestingStateFull is not reported as unused

**Comment by incendial** [comment]
@furaiev you mean, `TestingStateFull` is only used in `State<TestingStateFull>` and is not referenced by other files?

**Comment by furaiev** [comment]
@incendial yep, it is correct.

**Comment by dkrutskikh** [comment]
Available in 4.11.0 release 🚀
